fhir-quality-measure/src/main/java/com/ibm/fhir/ecqm/common/MeasureEvaluation.java

```diff
@@ -0,0 +1,702 @@
+                            newSDEItem.put(code, 1);
+                            sdeAccumulators.put(sdeAccumulatorKey, newSDEItem);
+                        } else {
+                            sdeAccumulators.get(sdeAccumulatorKey).put(code, 1);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private MeasureReportT processAccumulators(MeasureT measure, MeasureReportT report,
+            HashMap<String, HashMap<String, Integer>> sdeAccumulators,
+            List<MeasureSupplementalDataComponentT> sde, boolean isSingle, List<SubjectT> patients) {
+        sdeAccumulators.forEach((sdeKey, sdeAccumulator) -> {
+            sdeAccumulator.forEach((sdeAccumulatorKey, sdeAccumulatorValue) -> {
+
+                CodingT valueCoding = createCoding(sdeAccumulatorKey);
+                if (! sdeKey.equalsIgnoreCase("sde-sex")) {
+                    
+                    /** 
+                     * Match up the category part of our SDE key (e.g. sde-race has a category of race) with a 
+                     * patient extension of the same category (e.g. http://hl7.org/fhir/us/core/StructureDefinition/us-core-race)
+                     * and the same code as sdeAccumulatorKey (aka the value extracted from the CQL expression
+                     * named in the Measure SDE metadata) and then record the coding details.
+                     * 
+                     * We know that at least one patient matches the sdeAccumulatorKey or else it wouldn't show up
+                     * in the map.
+                     */
+                    
+                    String coreCategory = sdeKey.substring(sdeKey.lastIndexOf('-') >= 0 ? sdeKey.lastIndexOf('-') : 0);
+                    for( SubjectT pt : patients ) {
+                        valueCoding = getExtensionCoding(pt, coreCategory, sdeAccumulatorKey);
+                        
+                        //TODO - Is there any reason to continue looking at additional patients? The original 
+                        // cqf-ruler implementation would use the last matching patient's data vs. the first.
+                        if( valueCoding != null ) {
+                            break;
+                        }
+                    }
+                }
+                
+                ResourceT obs;
+                if (isSingle) {
+                    obs = createPatientObservation(measure, sdeKey, valueCoding);
+                } else {
+                    obs = createPopulationObservation(measure, sdeKey, valueCoding, sdeAccumulatorValue);
+                }
+                
+                addEvaluatedResource( report, obs );
+                addContained(report, obs);
+                //newRefList.add(new Reference("#" + this.getId.apply(obs)));
+                //report.addContained(obs);
+            });
+        });
+        //newRefList.addAll(report.getEvaluatedResource());
+        //report.setEvaluatedResource(newRefList);
```

Comment:
? what do the comments do that's useful downstream?

Reply:
This is a class that I brought forward from the cql-evaluator OSS project that I mentioned during review. I didn't spend any time trying to pretty it up. I certainly could if that is important to you folks though.